statistics/handle/handle_test.go

```diff
@@ -1266,18 +1266,24 @@ partition by range (a) (
 	c.Assert(s.do.StatsHandle().DumpStatsDeltaToKV(handle.DumpAll), IsNil)
 
 	tk.MustExec("set @@tidb_partition_prune_mode='static'")
-	tk.MustExec("analyze table t")
-	c.Assert(len(tk.MustQuery("show stats_meta").Rows()), Equals, 2) // p0 + p1
+	tk.MustExec("set @@tidb_analyze_version=1")
+	tk.MustExec("analyze table t") // both p0 and p1 are in ver1
+	c.Assert(len(tk.MustQuery("show stats_meta").Rows()), Equals, 2)
 
 	tk.MustExec("set @@tidb_partition_prune_mode='dynamic'")
 	tk.MustExec("set @@tidb_analyze_version=1")
-	err := tk.ExecToErr("analyze table t")
+	err := tk.ExecToErr("analyze table t") // try to build global-stats on ver1
 	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "[stats]: global statistics for partitioned tables only available in statistics version2, please set tidb_analyze_version to 2")
+	c.Assert(err.Error(), Equals, "[stats]: some partition level statistics are not in statistics version 2, please set tidb_analyze_version to 2 and analyze the this table")
 
+	tk.MustExec("set @@tidb_partition_prune_mode='dynamic'")
 	tk.MustExec("set @@tidb_analyze_version=2")
-	tk.MustExec("analyze table t")
-	c.Assert(len(tk.MustQuery("show stats_meta").Rows()), Equals, 3) // p0 + p1 + global
+	err = tk.ExecToErr("analyze table t partition p1") // only analyze p1 to let it in ver2 while p0 is in ver1
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[stats]: some partition level statistics are not in statistics version 2, please set tidb_analyze_version to 2 and analyze the this table")
+
+	tk.MustExec("analyze table t") // both p0 and p1 are in ver2
```

Comment:
Please add some tests to prove that when partition `p1` contains ver2-stats, we can use `analyze table t partition p0` to get the global-stats.

Reply:
OK, more cases have been added.